testutil/network/gatewaynet/network.go

```diff
@@ -0,0 +1,190 @@
+package gatewaynet
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	testcli "github.com/cosmos/cosmos-sdk/testutil/cli"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/network/basenet"
+	"github.com/pokt-network/poktroll/testutil/testkeyring"
+	"github.com/pokt-network/poktroll/x/application/client/cli"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+var _ network.InMemoryCosmosNetwork = (*inMemoryNetworkWithGateways)(nil)
+
+// inMemoryNetworkWithGateways is an implementation of the InMemoryCosmosNetwork interface.
+type inMemoryNetworkWithGateways struct {
+	//baseInMemoryNetwork basenet.BaseInMemoryCosmosNetwork
+	basenet.BaseInMemoryCosmosNetwork
+}
+
+// DefaultInMemoryNetworkConfig returns the default in-memory network configuration.
+// This configuration should sufficient populate on-chain objects to support reasonable
+// coverage around most session-oriented scenarios.
+func DefaultInMemoryNetworkConfig(t *testing.T) *network.InMemoryNetworkConfig {
+	t.Helper()
+
+	return &network.InMemoryNetworkConfig{
+		NumGateways:             5,
+		NumSuppliers:            2,
+		AppSupplierPairingRatio: 1,
+	}
+}
+
+// NewInMemoryNetworkWithGateways creates a new in-memory network with the given configuration.
+func NewInMemoryNetworkWithGateways(t *testing.T, cfg *network.InMemoryNetworkConfig) *inMemoryNetworkWithGateways {
+	t.Helper()
+
+	return &inMemoryNetworkWithGateways{
+		BaseInMemoryCosmosNetwork: *basenet.NewBaseInMemoryCosmosNetwork(
+			t, cfg, testkeyring.NewPreGeneratedAccountIterator(),
+		),
+	}
+}
+
+func (memnet *inMemoryNetworkWithGateways) DelegateAppToGateway(
```

Comment:
Can we add godoc comments for these functions

Reply:
Was planing on it. :wink: